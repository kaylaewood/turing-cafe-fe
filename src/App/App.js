import React, { Component } from 'react';
import './App.css';
import ReservationContainer from '../ReservationContainer/ReservationContainer';
import Form from '../Form/Form';
import { getReservations, postReservation } from '../apiCalls/apiCalls';

class App extends Component {
  constructor() {
    super()
    this.state = {
      reservations: []
    }
  }

  componentDidMount() {
    getReservations()
    .then(data => {
      this.setState({
        reservations: data
      })
    })
  }

  submitReservation = (name, date, time, number) => {
    postReservation(name, date, time, number)
    .then(response => {
      getReservations()
      .then(data => {
        this.setState({
          reservations: data
        })
      })
    })
  }

  render() {
    return (
      <div className="App">
        <h1 className='app-title'>Turing Cafe Reservations</h1>
        <div className='resy-form'>
          <Form
            submitReservation={this.submitReservation}
          />
        </div>
        <div className='resy-container'>
          <ReservationContainer
            reservations={this.state.reservations}
          />
        </div>
      </div>
    )
  }
}

export default App;
