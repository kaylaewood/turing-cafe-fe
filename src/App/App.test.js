import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  it('should match snapshot as expected', () => {
    const wrapper = shallow(<App />)

    expect(wrapper).toMatchSnapshot();
  })
})
