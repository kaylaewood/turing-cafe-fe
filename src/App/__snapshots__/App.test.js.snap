// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot as expected 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  >
    <Form
      submitReservation={[Function]}
    />
  </div>
  <div
    className="resy-container"
  >
    <ReservationContainer
      reservations={Array []}
    />
  </div>
</div>
`;
